models/user.rb

```diff
@@ -68,4 +63,11 @@ def to_h(for_user: false)
   def inspect
     "#{self.class.name} - id: #{id} name: #{name}"
   end
+
+  def validate
+    super
+    validates_unique(:name, :email, { message: 'is already registered' })
+    validates_format /^[^\s]+$/, :name, message: 'may not be empty'
+    validates_format /^[^@\s]+@[^@\s]+\.[^@\s]+$/, :email
```

Comment:
You may be able to use this, which is apparently a standard import

URI::MailTo::EMAIL_REGEXP

(since hand-rolled email regexes are tricky)

Reply:
I remember talking with Toby about the regex last year. tl;dr the simple one was good enough. But this one didn’t cross my path. Thanks.

Reply:
up to you, don't care